segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/TruncateOperation.java

```diff
@@ -113,7 +114,9 @@
                                                 }
 
                                                 // Remove read index block entries.
-                                                chunkedSegmentStorage.deleteBlockIndexEntriesForChunk(txn, streamSegmentName, oldStartOffset, segmentMetadata.getStartOffset());
+                                                if (!segmentMetadata.isStorageSystemSegment()) {
```

Comment:
Question: could this problem appear in other operations apart from truncate?

Reply:
For read/write this check is already there. This is the only one that I somehow missed and that went undetected. 
(Concat operation is illegal on system segments and will throw an assert).

I have added extra check to the unit tests on system tests to assert on any attempt to read from store.

Reply:
while today DefragmentOperation is only called inline during concat and concat is not valid operation on system segments.
In near future we may add background defrag. So to avoid future unintentional mistakes I added similar check for DefragmentOperation just as extra precaution.